adapters/verizonmedia/verizonmedia.go

```diff
@@ -35,68 +36,82 @@ func (a *VerizonMediaAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo
-	err := json.Unmarshal(request.Imp[0].Ext, &bidderExt)
-	if err != nil {
-		err = &errortypes.BadInput{
-			Message: "ext.bidder not provided",
-		}
-		errors = append(errors, err)
-		return nil, errors
+	reqs := make([]*adapters.RequestData, 0, len(request.Imp))
+	headers := http.Header{}
+
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	if request.Device != nil {
+		headers.Set("User-Agent", request.Device.UA)
 	}
-	var verizonMediaExt openrtb_ext.ExtImpVerizonMedia
-	err = json.Unmarshal(bidderExt.Bidder, &verizonMediaExt)
-	if err != nil {
-		err = &errortypes.BadInput{
-			Message: err.Error(),
+
+	for idx, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(imp.Ext, &bidderExt)
+		if err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: ext.bidder not provided", idx),
+			}
+			errors = append(errors, err)
+			continue
 		}
-		errors = append(errors, err)
-		return nil, errors
-	}
 
-	if verizonMediaExt.Dcn == "" {
-		err = &errortypes.BadInput{
-			Message: "Missing param dcn",
+		var verizonMediaExt openrtb_ext.ExtImpVerizonMedia
+		err = json.Unmarshal(bidderExt.Bidder, &verizonMediaExt)
+		if err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: %s", idx, err.Error()),
+			}
+			errors = append(errors, err)
+			continue
 		}
-		errors = append(errors, err)
-		return nil, errors
-	}
 
-	if verizonMediaExt.Pos == "" {
-		err = &errortypes.BadInput{
-			Message: "Missing param pos",
+		if verizonMediaExt.Dcn == "" {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: missing param dcn", idx),
+			}
+			errors = append(errors, err)
+			continue
 		}
-		errors = append(errors, err)
-		return nil, errors
-	}
 
-	siteCopy := *request.Site
-	request.Site = &siteCopy
-	if err := changeRequestForBidService(request, &verizonMediaExt); err != nil {
-		errors = append(errors, err)
-		return nil, errors
-	}
+		if verizonMediaExt.Pos == "" {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: missing param pos", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
 
-	reqJSON, err := json.Marshal(request)
-	if err != nil {
-		errors = append(errors, err)
-		return nil, errors
-	}
+		// Split up multi-impression requests into multiple requests so that
+		// each split request is only associated to a single impression
+		reqCopy := *request
+		reqCopy.Imp = []openrtb.Imp{imp}
 
-	thisURI := a.URI
+		siteCopy := *request.Site
+		reqCopy.Site = &siteCopy
 
-	headers := http.Header{}
-	headers.Add("Content-Type", "application/json;charset=utf-8")
-	headers.Add("Accept", "application/json")
-	headers.Set("User-Agent", request.Device.UA)
-	headers.Add("x-openrtb-version", "2.5")
+		if err := changeRequestForBidService(&reqCopy, &verizonMediaExt); err != nil {
+			errors = append(errors, err)
+			return nil, errors
```

Comment:
@Aadeshp , are we sure we want to return because this particular Imp does not pass the checks found in `changeRequestForBidService()`? Don't we want to `continue` instead? The `changeRequestForBidService()` function pretty much makes sure this `Imp` is a banner and check for its dimensions. Are sure we want to exit at the first non `banner` or non-properly dimensioned `banner` we find?

Reply:
ah yes good find!

Reply:
fixed!